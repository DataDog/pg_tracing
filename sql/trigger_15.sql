-- Test explain on a before trigger
-- This will go through ExecutorEnd without any parent executor run
/*dddbs='postgres.db',traceparent='00-fed00000000000000000000000000001-0000000000000003-01'*/ explain INSERT INTO before_trigger_table VALUES(10);
SELECT trace_id, span_type, span_operation, lvl from peek_ordered_spans where trace_id='fed00000000000000000000000000001';

CALL clean_spans();
