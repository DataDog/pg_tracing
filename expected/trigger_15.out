-- Test explain on a before trigger
-- This will go through ExecutorEnd without any parent executor run
/*dddbs='postgres.db',traceparent='00-fed00000000000000000000000000001-0000000000000003-01'*/ explain INSERT INTO before_trigger_table VALUES(10);
                            QUERY PLAN                            
------------------------------------------------------------------
 Insert on before_trigger_table  (cost=0.00..0.01 rows=0 width=0)
   ->  Result  (cost=0.00..0.01 rows=1 width=4)
(2 rows)

SELECT trace_id, span_type, span_operation, lvl from peek_ordered_spans where trace_id='fed00000000000000000000000000001';
             trace_id             |   span_type    |                    span_operation                    | lvl 
----------------------------------+----------------+------------------------------------------------------+-----
 fed00000000000000000000000000001 | Utility query  | explain INSERT INTO before_trigger_table VALUES(10); |   0
 fed00000000000000000000000000001 | ProcessUtility | ProcessUtility                                       |   1
 fed00000000000000000000000000001 | Insert query   | explain INSERT INTO before_trigger_table VALUES($1); |   2
 fed00000000000000000000000000001 | Planner        | Planner                                              |   3
(4 rows)

CALL clean_spans();
